File: backend/app/schemas/watchlist.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field


class WatchlistCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    objective: str = Field(pattern="^(GROWTH|VALUE|STABILITY)$")


class WatchlistUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=100)
    objective: str | None = Field(default=None, pattern="^(GROWTH|VALUE|STABILITY)$")


class WatchlistOut(BaseModel):
    id: UUID
    name: str
    objective: str
    last_viewed_at: datetime | None
    created_at: datetime

    class Config:
        from_attributes = True


class WatchlistItemCreate(BaseModel):
    instrument_id: UUID

File: frontend/src/types/watchlist.ts
import { Objective } from './market';

export interface WatchlistCreate {
  name: string;
  objective: Objective;
}

export interface WatchlistUpdate {
  name?: string;
  objective?: Objective;
}

export interface WatchlistOut {
  id: string;
  name: string;
  objective: Objective;
  last_viewed_at: string | null;
  created_at: string;
}

export interface StockAdd {
  ticker: string;
}

export interface StockOut {
  instrument_id: string;
  ticker: string;
  name: string;
  exchange: string;
  added_at: string;
}
